### ISSUE TITLE
[SQLancer] [YSQL] Trying to drop serial PK column leads to failed inserts

### ISSUE THREAD

**Issue by qvad** [yugabyte/yugabyte-db#11110]
### Description

SQL scenario is attached. 
If we exclude ALTER TABLE command from scenario everything is fine. And query itself is failed with execution, but looks like we commit partial results in this case

```
CREATE TABLE t2(c0 TEXT , c1 serial  PRIMARY KEY);
ALTER TABLE t2 DROP column c1;
-- ERROR: Invalid argument: cannot remove a key column
INSERT INTO t2(c0) VALUES('ABC')
-- ERROR: Not found: Unable to find relation for sequence 20992
```

**Comment by qvad** [comment]
PG11 on the other side, allows ALTER TABLE for dropping PK column.
Here is output from it
```
postgres.public> CREATE TABLE t2(c0 TEXT , c1 serial  PRIMARY KEY)
[2022-01-17 15:57:12] completed in 8 ms
postgres.public> ALTER TABLE t2 DROP column c1
[2022-01-17 15:57:12] completed in 3 ms
postgres.public> INSERT INTO t2(c0) VALUES('ABC')
[2022-01-17 15:57:12] 1 row affected in 3 ms
```

**Comment by yifanguan** [comment]
Issue resolved by commit b99a04644de1eb7ab98a494976f2f7b090c2ef5f
